dataactbroker/helpers/validation_helper.py

```diff
@@ -126,6 +209,23 @@ def derive_unique_id(row, is_fabs):
     return unique_id
 
 
+def derive_unique_id_vectorized(frame: pd.DataFrame, is_fabs):
+    """ Derives the unique ID of a row and puts it in the proper format for the error/warning report.
+
+        Args:
+            frame: the pd.DataFrame whose 'unique_id' column's values will be updated
+            is_fabs: a boolean indicating if the submission is a FABS submission or not
+
+        Returns:
+            A Series with properly formatted unique ID all rows depending on if it's a FABS or DABS submission
+    """
+    if not is_fabs:
+        return 'TAS: ' + frame['display_tas'][frame['display_tas'].notnull()].astype(str)
+    else:
+        return 'AssistanceTransactionUniqueKey: ' + \
+               frame['afa_generated_unique'][frame['afa_generated_unique'].notnull()].astype(str)
```

Comment:
From what I could tell briefly reading the docs on Pandas, string concatenation of a column (a whole Series) and not just a value, is done like this using `+`. I couldn't find explicit docs on comparing to .format, but I was afraid it would "un"-vectorize it if causing each element to be passed to a string-formatting function. 

My guess is that pandas has overloaded the `+` operator on strings to do some kind of "add this to all strings in the Series" in one vectorized operation.

To be safe, and get the most out of performance I think we should stick to this kind of string concat until we can find docs that say how using formatting / interpolation is ok.

Reply:
Works for me 👍

Reply:
I haven't heard about a security concern, just a (very tiny) performance hit